### PR TITLE
base commits

### PR DESCRIPTION
**Pull request by cjus**

### CODE REVIEW

**Comment by cjus** [comment]
Sample moose log entries:

```
32:28 INFO  c83f6 moose_cli::cli::routines] current_schema_hash: "UserActivity_0_1"-"9cd233cab74d1cb6c06e34c07becc160e78ccc0f8fba113f433a0573d87a1675"
32:28 INFO  c83f6 moose_cli::cli::routines] prev_schema_hash: "UserActivity_0_0"-"b0d7ebf212ee1b7753ea3e5c89b993278a8b3ea4c4acdb9590d1efbde27da244"
32:28 INFO  c83f6 moose_cli::cli::routines] table_schema_hash: "UserActivity_0_0"-"b0d7ebf212ee1b7753ea3e5c89b993278a8b3ea4c4acdb9590d1efbde27da244"
32:28 INFO  c83f6 moose_cli::cli::routines] table_schema_hash: "UserActivity_0_1"-"9cd233cab74d1cb6c06e34c07becc160e78ccc0f8fba113f433a0573d87a1675"
```

**Comment by cjus** [comment]
```
#[derive(Debug, Clone)]
pub struct FrameworkObject {
    pub data_model: DataModel,
    pub table: ClickHouseTable,
    pub topic: String,
    pub ts_interface: TypescriptInterface,
    pub original_file_path: PathBuf,
}
```

File: apps/framework-cli/src/infrastructure/olap/clickhouse.rs [hunk 1]
```diff
@@ -205,6 +205,68 @@ pub async fn check_is_table_new(
...
+    let query = format!("SELECT name, type FROM system.columns WHERE (database = '{db_name}' AND table = '{table_name}')");
+    let mut cursor = client.query(query.as_str()).fetch::<(String, String)>()?;
+
+    let mut columns = vec![];
+
+    while let Some((name, column_type)) = cursor.next().await? {
+        columns.push((name, column_type));
+    }
+
+    columns.sort();
```
Comment (phiSgr) [inline apps/framework-cli/src/infrastructure/olap/clickhouse.rs:249]: if the ordering of fields changed, we should recreate the table.

**Comment by phiSgr** [comment]
BTW, we should probably `panic` in production if the columns in DB don't match the columns in the schema files.

**Pull request closed without merging**